controllers/cloudstackmachine_controller.go

```diff
@@ -70,93 +74,95 @@ func (r *CloudStackMachineReconciler) Reconcile(ctx context.Context, req ctrl.Re
-			if err := patchHelper.Patch(ctx, csMachine); retErr == nil && err != nil {
-				retErr = err
+			if err = patchHelper.Patch(ctx, csMachine); err != nil {
+				msg := "error patching CloudStackMachine %s/%s"
+				err = errors.Wrapf(err, msg, csMachine.Namespace, csMachine.Name)
+				retErr = multierror.Append(retErr, err)
 			}
 		}()
 	}
 
 	// Fetch the CAPI Machine.
-	machine, retErr := util.GetOwnerMachine(ctx, r.Client, csMachine.ObjectMeta)
-	if retErr != nil {
-		return ctrl.Result{}, retErr
-	} else if machine == nil {
+	capiMachine, err := util.GetOwnerMachine(ctx, r.Client, csMachine.ObjectMeta)
+	if err != nil {
+		return ctrl.Result{}, err
+	} else if capiMachine == nil {
 		log.Info("Waiting for CAPI cluster controller to set owner reference on CloudStack machine.")
 		return ctrl.Result{RequeueAfter: RequeueTimeout}, nil
 	}
 
 	// Fetch the CAPI Cluster.
-	cluster, retErr := util.GetClusterFromMetadata(ctx, r.Client, machine.ObjectMeta)
-	if retErr != nil {
+	cluster, err := util.GetClusterFromMetadata(ctx, r.Client, capiMachine.ObjectMeta)
+	if err != nil {
 		log.Info("Machine is missing cluster label or cluster does not exist.")
 		return ctrl.Result{RequeueAfter: RequeueTimeout}, nil
 	}
 
-	// Check the machine is not paused.
+	// Check if the machine is paused.
 	if annotations.IsPaused(cluster, csMachine) {
 		log.Info("CloudStackMachine or linked Cluster is paused. Requeuing reconcile.")
 		return reconcile.Result{}, nil
 	}
 
-	// Delete VM instance if deletion timestamp present.
-	if !csMachine.DeletionTimestamp.IsZero() {
-		return r.reconcileDelete(log, csMachine)
-	}
-
 	// Fetch the CloudStack cluster associated with this machine.
 	csCluster := &infrav1.CloudStackCluster{}
-	if retErr := r.Client.Get(
+	if err := r.Client.Get(
 		ctx,
 		client.ObjectKey{
 			Namespace: csMachine.Namespace,
 			Name:      cluster.Spec.InfrastructureRef.Name},
-		csCluster); retErr != nil {
-		if client.IgnoreNotFound(retErr) == nil {
+		csCluster); err != nil {
+		if client.IgnoreNotFound(err) == nil {
 			log.Info("CloudStackCluster not found.")
 			return ctrl.Result{RequeueAfter: RequeueTimeout}, nil
 		} else {
-			return ctrl.Result{}, retErr
+			return ctrl.Result{}, err
 		}
 	} else if csCluster.Status.ZoneID == "" {
 		log.Info("CloudStackCluster ZoneId not initialized. Likely not ready.")
 		return ctrl.Result{RequeueAfter: RequeueTimeout}, nil
 	}
 
+	// Delete VM instance if deletion timestamp present.
+	if !csMachine.DeletionTimestamp.IsZero() {
+		return r.reconcileDelete(ctx, log, csMachine, capiMachine, csCluster)
+	}
+
 	// Reconcile a VM instance for creates/updates
-	return r.reconcile(log, ctx, csCluster, csMachine, machine)
+	return r.reconcile(ctx, log, csMachine, capiMachine, csCluster)
 }
 
 // Actually reconcile/Create a VM instance.
 func (r *CloudStackMachineReconciler) reconcile(
-	log logr.Logger,
 	ctx context.Context,
-	csCluster *infrav1.CloudStackCluster,
+	log logr.Logger,
```

Comment:
why are we passing this around? Should it just be saved at the Reconciler level and referenced as r.log?

Reply:
I like what you're thinking. That said, I actually want to create scope structs(or maps) that are passed to the cloud package that would contain the log and all other vars. So, I'm leaving the log object encapsulation until/if we get around to that.

We will eventually want to have the log in the cloud package.